controllers/vmprometheusconverter_controller.go

```diff
@@ -235,17 +235,27 @@ func waitForAPIResource(ctx context.Context, client discovery.DiscoveryInterface
 	for {
 		select {
 		case <-tick.C:
-			_, apiLists, err := client.ServerGroupsAndResources()
+			groups, err := client.ServerGroups()
 			if err != nil {
-				l.Error(err, "cannot get  server resource")
+				l.Error(err, "cannot get server groups")
+				continue
 			}
-			for _, apiList := range apiLists {
-				if apiList.GroupVersion == apiGroupVersion {
-					for _, r := range apiList.APIResources {
-						if r.Kind == kind {
-							l.Info("api resource is ready")
-							return nil
+			for _, g := range groups.Groups {
+				for _, v := range g.Versions {
+					if v.GroupVersion != apiGroupVersion {
+						continue
+					}
```

Comment:
I think we don't need this check at all, it doesn't provide extra info than `client.ServerResourcesForGroupVersion(apiGroupVersion)`. 
`apiGroupVersion` is already passed by caller and can be used in `api, err := client.ServerResourcesForGroupVersion(apiGroupVersion)` directly.

Reply:
The reason for advance calling ServerGroups is that calling `ServerResourcesForGroupVersion` for a group missing in the cluster causes an error - the point of this piece of code is to wait for the group to appear in the cluster.

Reply:
> wait for the group to appear in the cluster.

If group is not there yet, `client.ServerResourcesForGroupVersion` should also give error, no?

Reply:
> If group is not there yet, client.ServerResourcesForGroupVersion should also give error, no?

Generally yes, but in this case `ServerResourcesForGroupVersion` will be called only for those groups returned by `ServerGroups`, i.e. known to exist in the cluster.

Reply:
hmm, I still don't see the harm of calling `ServerResourcesForGroupVersion` directly. 
If `monitoring.coreos.com` apiGroup is not exist, we can just ignore the error of `ServerResourcesForGroupVersion`, as user may not using prometheus CRDs. 
```
		case <-tick.C:
			api, err := client.ServerResourcesForGroupVersion(apiGroupVersion)
			if err != nil {
				if !strings.Contains(err.Error(), "could not find the requested resource") {
					l.Error(err, "cannot get server resource for api group version")
				}
				continue
			}
			for _, r := range api.APIResources {
				if r.Kind != kind {
					continue
				}
				l.Info("api resource is ready")
				return nil
			}
```

Reply:
I'm not a fan of using errors in this way to create business logic, but it would make the code simpler, ok, I'll make the fix and test it today.

Reply:
Fixed